include/PairCache.h

```diff
@@ -23,78 +23,87 @@
 #include "Utils.h"
 #include "tables/DBTableBase.h"
 
+/*
+ * Key1 -> Key2 - OneToOne relation
+ * Key2 -> Key1 - OneToMany relation
+*/
+template<typename Key1, typename Key2, typename Key1Value>
 class PairCache {
```

Comment:
I think this cache rightnow is only used for DatasetProjects, so i guess let's skip all the generics and make concrete, rename it to DatasetProjectCache and rename the methods accordingly to make it more readable

Reply:
fixed